mono/utils/mono-dl.c

```diff
@@ -143,6 +148,26 @@ get_dl_name_from_libtool (const char *libtool_file)
 	return line;
 }
 
+#ifdef ENABLE_NETCORE
+static const char *
+fix_libc_name (const char *name)
+{
+	if (strncmp (name, "libc", 4) == 0) {
+		// Taken from CoreCLR: https://github.com/dotnet/coreclr/blob/6b0dab793260d36e35d66c82678c63046828d01b/src/pal/src/loader/module.cpp#L568-L576
+#if defined (HOST_DARWIN)
+		return "/usr/lib/libc.dylib";
+#elif defined (__FreeBSD__)
+		return "libc.so.7";
+#elif defined (LIBC_SO)
+		return LIBC_SO;
```

Comment:
Once we have netcore on Android we should check that this define works there.

Reply:
It does. The same code is used by CoreCLR and it was already compiled on Android few times.